test/doc/one-time-setup.md

```diff
@@ -0,0 +1,214 @@
+az iot hub create --resource-group {resource group name} --name {IoT hub name} 
+~~~
+
+Note down the `event hub compatible endpoint` and the primary connection string of the `iothubowner` policy. These will need to be set in the `E2E_EVENT_HUB_ENDPOINT` and `E2E_IOT_HUB_CONNECTION_STRING` environment variables later to run the tests.
+
+You can get these via the CLI by running 
+~~~sh
+# Primary Connection string
+az iot hub connection-string show --hub-name {IoT hub name} --key-type primary
+
+#Default eventhub compatible end point
+az iot hub connection-string show --hub-name {IoT hub name} --default-eventhub
+
+~~~
+##### Device Provisioning Service (DPS)
+Create a DPS instance. A subset of the end-to-end tests will use DPS for testing device provisioning scenarios. Note down the `ID Scope` of this DPS instance. You will need to set it in the `dpsIdScope` configuration variable later to run the tests.
+
+Using the CLI,
+~~~sh
+az iot dps create --name {dps group name} --resource-group {resource group name} --location {region}
+
+# To get the ID Scope
+az iot dps show --name {dps group name} | grep idScope
+~~~
+
+###### Create enrollment group for symmetric key based enrollment
+In your DPS instance, create a new enrollment group for symmetric key based enrollment. For this enrollment group, set the attestation type to be symmetric key, set the IoTEdge Device setting to true, and link the group to your IoT hub with the access policy of `iotHubOwner`. 
+
+Using the CLI,
+~~~sh
+#Link the DPS group to your Iot hub
+az iot dps linked-hub create --dps-name {dps group name} --resource-group {resource group name} \
+  --connection-string "{Iot hub ConnectionString}" --location westus2
+
+#Create enrollment group for symmetric key based enrollment
+ az iot dps enrollment-group create -g {resource group name} --dps-name {dps group name} \
+  --enrollment-id {symkey enrollment group name } --edge-enabled
+~~~
+You do not have to create the symmetric key. The system will auto generate it for you when the enrollment group is created. After the enrollment group is created, note down the Primary Key. You will need to set this in the `E2E_DPS_GROUP_KEY` environment variable later to run the tests.
+
+See [Symmetric key attestation](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-auto-provision-symmetric-keys?view=iotedge-2020-11&tabs=windows) for further details.
+
+###### Create enrollment group for X.509 certificate based enrollment
+In your DPS instance, create another enrollment group for X.509 certificate based enrollment. For this enrollment group, set the attestation type to be certificate and upload the root CA certificate that you had created earlier, set the IoTEdge Device setting to true, and link the group to your IotHub with the access policy of `iotHubOwner`.
+
+~~~sh
+# Upload the root ca cert and set it to be verified
+az iot dps certificate create --certificate-name {dps root ca name}  --resource-group {resource group name} \
```

Comment:
We did talk about this already, putting it here for convenience:

A comment about the minimum versions of Azure CLI needed might be beneficial considering that both Noel & I ran into issues here.

need az cli version 2.28.0 and azure-iot  version 0.10.17 at a minimum